_posts/2017-04-06-git-aliases-ideas-from-our-setups.md

```diff
@@ -0,0 +1,72 @@
+---
+date: 2017-04-06
+title: "Git aliases: Ideas from our setups"
+categories:
+  - CLI
+  - Git
+  - Alias
+author_staff_member: 02_fausto
+medium_link: https://blabla.com
+---
+
+In this post we will keep an updated list of the Git aliases we use the most, particularly the ones we find interesting (i.e., not `c = commit`) and that have increased our productivity using Git.
+
+# Rebasing
+
+For the following aliases we're going to define shell functions inside of an alias and call them directly afterwards.
+
+## rbb
+
+`rbb = "!f() { git rebase -i HEAD~$1; }; f"`
+
+Interactive rebase by the amount of commits passed through the first argument. Using `git rbb 3` in a project with at least three commits will leave you in your editor of choice with something like this:
+
+![interactive rebase alias]({{ site.url }}/images/git_aliases/rbb.jpg)
+
+## cfix
+
+We can also use backticks surrounding a command to evaluate with the shell and pass the results as a part of our git alias. We use this with `rev-parse` in the following alias:
+
+``cfix = "!f() { git commit --fixup `git rev-parse --short HEAD` && git rebase -i HEAD~2 -i --autosquash; }; f"``
+
+We use `cfix` when the following situation arises. Say you've committed something with the message `Fix the terrible bug` only to find out that the fix did not actually work the way you expected. So you make your changes and go:
+
+`git add .`
+
+Now you commit the new, real fix:
+
+`git commit -m "Fix the terrible bug... again"`
+
+Wouldn't it be nice to just have the new changes feature in the previous commit as one? Well, you can do that with a `rebase`:
```

Comment:
I see what you mean with the first sentence but I think it's not completely clear.